update.ps1

```diff
@@ -1,233 +1,48 @@
-$p = $person | ConvertFrom-Json
-$aRef = $AccountReference | ConvertFrom-Json
-$success = $false
-$auditLogs = [System.Collections.Generic.List[PSCustomObject]]::new()
-
-#Generate surname conform nameconvention
-function Get-LastName {
-    [CmdletBinding()]
-    param (
-        [Parameter(Mandatory)]
-        [ValidateNotNullOrEmpty()]
-        [object]
-        $person
-    )
-
-    if ([string]::IsNullOrEmpty($person.Name.FamilyNamePrefix)) {
-        $prefix = ""
-    }
-    else {
-        $prefix = $person.Name.FamilyNamePrefix + " "
-    }
-
-    if ([string]::IsNullOrEmpty($person.Name.FamilyNamePartnerPrefix)) {
-        $partnerPrefix = ""
-    }
-    else {
-        $partnerPrefix = $person.Name.FamilyNamePartnerPrefix + " "
-    }
-
-    $Surname = switch ($person.Name.Convention) {
-        "B" { $person.Name.FamilyName }
-        "BP" { $person.Name.FamilyName + " - " + $partnerprefix + $person.Name.FamilyNamePartner }
-        "P" { $person.Name.FamilyNamePartner }
-        "PB" { $person.Name.FamilyNamePartner + " - " + $prefix + $person.Name.FamilyName }
-        default { $prefix + $person.Name.FamilyName }
-    }
-
-    $Prefix = switch ($p.Name.Convention) {
-        "B" { $prefix }
-        "BP" { $prefix }
-        "P" { $partnerPrefix }
-        "PB" { $partnerPrefix }
-        default { $prefix }
-    }
-    $output = [PSCustomObject]@{
-        surname  = $Surname
-        prefixes = $Prefix.Trim()
-    }
-    Write-Output $output
-}
-
-# Script Configuration
-$departmentLookupProperty = { $_.Department.ExternalId }
-$titleLookupProperty = { $_.Title.ExternalId }
-
-# Employee Account mapping
-$account = [PSCustomObject]@{
-    staffnumber          = $p.ExternalId
-    firstname            = $p.Name.NickName
-    lastname             = (Get-LastName -Person $p).surname
-    middlename           = (Get-LastName -Person $p).prefixes    
-    initials             = $p.Name.Initials
-    email                = $p.Contact.Business.Email
-    phone                = $p.Contact.Business.Phone.Fixed
-    dateofbirth          = if ($null -ne $p.Details.BirthDate ) { '{0:yyyy-MM-dd}' -f ([datetime]$p.Details.BirthDate) };
-    startdate            = if ($null -ne $p.PrimaryContract.StartDate ) { '{0:yyyy-MM-dd}' -f ([datetime]$p.PrimaryContract.StartDate) };
-    enddate              = $null
-    positionsPerOrgUnits = @()
-}
+# PowerShell V2
+# Version: 1.0.0
+#################################################
 
 # Enable TLS1.2
 [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::Tls12
 
 # Set debug logging
-switch ($($config.IsDebug)) {
+switch ($($actionContext.Configuration.isDebug)) {
     $true { $VerbosePreference = 'Continue' }
     $false { $VerbosePreference = 'SilentlyContinue' }
 }
 
+# Script Configuration
+$departmentLookupProperty = { $_.Department.ExternalId }
+$titleLookupProperty = { $_.Title.ExternalId }
```

Comment:
This part of the code is complex. Not for you to change. This should be something for the connector team.